ui/js/models/resultset.js

```diff
@@ -269,35 +271,66 @@ treeherder.factory('ThResultSetModel', ['$rootScope', '$http', '$location',
-                        $q.resolve().then(function () {
+                        $q.resolve().then(() => {
                             if (tclabels.length === 0) {
                                 return;
                             }
-                            let url = queue.buildSignedUrl(queue.getLatestArtifact, decisionTaskID, 'public/action.yml');
-                            return $http.get(url).then(function (resp) {
-                                let action = resp.data;
-                                let template = $interpolate(action);
-                                let taskLabels = tclabels.join(',');
-                                action = template({
-                                    action: 'add-tasks',
-                                    action_args: `--decision-id ${decisionTaskID} --task-labels ${taskLabels}`,
+
+                            let actionTaskId = tc.slugid();
+                            let url = queue.buildSignedUrl(queue.getLatestArtifact, decisionTaskID, 'public/actions.json');
+
+                            return $http.get(url).then((resp) => {
+                                // Use action.yml if it's the wrong version of actions.json
+                                if (resp.data.version !== 1) {
+                                    url = queue.buildSignedUrl(queue.getLatestArtifact, decisionTaskID, 'public/action.yml');
+
+                                    return $http.get(url).then((resp) => {
+                                        let action = resp.data;
+                                        let template = $interpolate(action);
+                                        let taskLabels = tclabels.join(',');
+
+                                        action = template({
+                                            action: 'add-tasks',
+                                            action_args: `--decision-id ${decisionTaskID} --task-labels ${taskLabels}`,
+                                        });
+
+                                        let actionTask = thTaskcluster.refreshTimestamps(jsyaml.safeLoad(action));
+
+                                        return queue.createTask(actionTaskId, actionTask);
+                                    });
+                                }
+
+                                const action = resp.data.actions.filter(action => action.name === 'add-new-jobs')[0];
+                                const staticActionVariables = resp.data.variables;
+
+                                return $http.get(`https://queue.taskcluster.net/v1/task/${decisionTaskID}`).then((resp) => {
+                                    const originalTask = resp.data;
+
+                                    let actionTask = actionsRender(action.task, _.defaults({}, {
+                                        taskGroupId: originalTask.taskGroupId,
+                                        taskId: decisionTaskID,
+                                        task: originalTask,
```

Comment:
@helfi92 I think the `originalTask` is supposed to be the task definition of the task that this was called on and not the corresponding decision task.